src/utilities.jl

```diff
@@ -23,22 +23,66 @@ function tensor_create{order, dim}(::Type{SymmetricTensor{order, dim}}, f)
     return ex
 end
 
-# create recursive muladd exp from two expression arrays
-function make_muladd_exp(ex1, ex2)
+# reduced two expressions by summing the products
+function reducer(ex1i, ex2i)
+    ex1, ex2 = remove_duplicates(ex1i, ex2i)
     N = length(ex1)
-    ex = Expr(:call)
-    exn = Expr(:call, :*, ex1[1], ex2[1])
+    expr = :($(ex1[1]) * $(ex2[1]))
 
-    if N == 1 # return only the multiplication
-        return exn
+    for i in 2:N
+        expr = :($(expr) + $(ex1[i]) * $(ex2[i]))
+        # expr = :(muladd($(ex1[i]), $(ex2[i]), $expr)) # faster on some architectures!?
```

Comment:
This causes the regression for `dcontract` (which also is the reason for the AD benchmarks most likely). Before this PR we used `muladd` instructions but that turned out to be slower for `dot` and `norm`. It is really weird when `muladd` is faster or not. I will try to investigate this more and perhaps we can tailor the use of `muladd` a bit better.